### PR TITLE
Pixhawk 2 mag fix

### PR DESCRIPTION
**Pull request by mhkabir**
Adds support for a 4th magnetometer (Pixhawk 2 has 3 internal + atleast 1 external in GPS unit)

@dagar Would be great if you could crosstest with Pixhawk 2 + external mag. Wipe everything and recalibrate.

### CODE REVIEW

File: ROMFS/px4fmu_common/init.d/rc.sensors [hunk 1]
```diff
@@ -135,10 +135,13 @@ then
 		then
 		fi
 
+		# We don't start this mag on the Pixhawk 2, because we do not realistically
```
Comment (LorenzMeier) [inline ROMFS/px4fmu_common/init.d/rc.sensors]: This is also a 2nd accel. Isn't the right fix to change the start order such that the external units come first?

**Comment by LorenzMeier** [comment]
Needs HW testing but code looks good. Nice job!

File: src/modules/sensors/common.h [hunk 2]
```diff
@@ -43,6 +43,14 @@
...
-static const int SENSOR_COUNT_MAX = 3;
+constexpr uint8_t MAG_COUNT_MAX = 4;
+constexpr uint8_t GYRO_COUNT_MAX = 3;
+constexpr uint8_t ACCEL_COUNT_MAX = 3;
+constexpr uint8_t BARO_COUNT_MAX = 3;
+
+constexpr uint8_t SENSOR_COUNT_MAX = math::max(MAG_COUNT_MAX,
+				     math::max(GYRO_COUNT_MAX,
+						     math::max(ACCEL_COUNT_MAX,
+								     BARO_COUNT_MAX)));
```
Comment (mhkabir) [inline src/modules/sensors/common.h]: @LorenzMeier @dagar Any way to fix this astyle fail?

**Comment by mhkabir** [comment]
We're good to go!

**Pull request merged**